packages/nimble-components/src/rich-text-editor/styles.ts

```diff
@@ -146,6 +178,29 @@ export const styles = css`
         border-top-color: rgba(${borderRgbPartialColor}, 0.1);
         height: var(--ni-private-rich-text-editor-footer-section-height);
         overflow: hidden;
+        visibility: visible;
+    }
+
+    :host([footer-hidden]) .footer-section {
+        visibility: hidden;
+        height: 0px;
+    }
+
+    :host([footer-hidden]) .ProseMirror {
+        ${
+            /**
+             * Minimum height is the combination of existing minimum height of the tiptap div, footer height and extra padding added to the tiptap div.
+             * With this calculation, the editor will extend to use the footer height when it is hidden.
+             *
+             * Use case: If the footer is initially hidden and is dynamically enabled when the editor is focused, there will be no layout shift,
+             * and the footer will smoothly appear within the editor.
+             */ ''
+        }
+        min-height: calc(var(--ni-private-rich-text-editor-editor-minimum-height) + var(--ni-private-rich-text-editor-footer-section-height) + 4px);
```

Comment:
What is the additional `4 px` here? Buffer?

Reply:
This `4px` should not come here. Previously, I added the minimum height of the editor section as `32px` but for a minimum one-line height, it should be `36px`. Font size of paragraph (18px) + Padding (top 8px + bottom 8px) + Border (top 1px + bottom 1px) = `36px`. 

Therefore, I removed the `4px` from the calculation.